covidthailand.py

```diff
@@ -2597,794 +2601,1157 @@ def topprov(df, metricfunc, valuefunc=None, name="Top 5 Provinces", num=5, other
-            "PCR Tests per Case",
-            "PUI per Case",
-            "PUI*3 per Case",
-            "PUI per Walkin Case",
+    # fig, ax = plt.subplots(figsize=[20, 10])
+    # df['2020-12-12':].plot(
+    #     ax=ax,
+    #     y=cols[:1],
+    #     linewidth=5,
+    #     title=title
+    # )
+    # df['2020-12-12':].plot(
+    #     ax=ax,
+    #     y=cols[1:],
+    #     colormap=plt.cm.Set1,
+    # )
+    # # df['2020-12-12':].plot(
+    # #     ax=ax,
+    # #     linestyle='--',
+    # #     y=un_ma,
+    # # )
+    # ax.legend(legend)
+    # plt.tight_layout()
+    # plt.savefig('outputs/positivity_2.png')
+
+    plot_area(df=df, png_prefix='positivity', cols_subset=cols,
+              title='Positive Rate: Is enough testing happening?', legends=legends,
+              kind='line', stacked=False, percent_fig=False, ma=True, cmap='tab10')
+
+
+
+    df['PUI per Case'] = df['Tested PUI (MA)'].divide(df['Cases (MA)'])
+    df['PUI3 per Case'] = df['Tested PUI (MA)']*3 / df['Cases (MA)']
+    df['PUI3 per Walkin'] = df['Tested PUI (MA)']*3 / df['Cases Walkin (MA)']
+    df['PUI per Walkin'] = df['Tested PUI (MA)'].divide( df['Cases Walkin (MA)'] )
+    df['Tests per case'] = df['Tests Corrected+Private (MA)'] / df['Cases (MA)']
+    df['Tests per positive'] = df['Tests Corrected+Private (MA)'] / df['Pos Corrected+Private (MA)']
+
+    # fig, ax = plt.subplots(figsize=[20, 10])
+    # df['2020-12-12':].plot(
+    #     ax=ax,
+    #     kind='line',
+    #     y=[
+    #         'Tests per positive',
+    #     ],
+    #     linewidth = 7,
+    # )
+    # df['2020-12-12':].plot(
+    #     ax=ax,
+    #     use_index=True,
+    #     kind='line',
+    #     colormap='Set1',
+    #     y=[
+    #         'Tests per case',
+    #         'PUI per Case',
+    #         'PUI3 per Case',
+    #         'PUI per Walkin',
+    #     ],
+    #     title='Thailand Tests per Confirmed Case\n'
+    #     '(7 day rolling mean)\n'
+    #     f'Updated: {TODAY().date()}\n'
+    #     'djay.github.io/covidthailand'
+    # )
+    # ax.legend(
+    #     [
+    #         'PCR Tests per Positive',
+    #         'PCR Tests per Case',
+    #         'PUI per Case',
+    #         'PUI*3 per Case',
+    #         'PUI per Walkin Case',
+    #     ]
+    # )
+    # plt.tight_layout()
+    # plt.savefig('outputs/tests_per_case.png')
+
+    cols = ['Tests per positive', 'Tests per case', 'PUI per Case', 'PUI3 per Case', 'PUI per Walkin']
+    legends = [
+            'PCR Tests per Positive',
+            'PCR Tests per Case',
+            'PUI per Case',
+            'PUI*3 per Case',
+            'PUI per Walkin Case',
         ]
-    )
-    plt.tight_layout()
-    plt.savefig("outputs/tests_per_case.png")
-
+    plot_area(df=df, png_prefix='tests_per_case', cols_subset=cols,
+              title='Thailand Tests per Confirmed Case', legends=legends,
+              kind='line', stacked=False, percent_fig=False, ma=True, cmap='tab10')
```

Comment:
I think other than "highlight_cols", we could also add a "main_data_cols".... not sure on the name. But something that we work out the data of last significant data. For example in this plot the positive rate from testing data is the main thing and the others are there as predictions. It would be nice if the title clearly stated that the testing data wasn't updated since 2021-04-24. Something like "Last Change"? "Last Data update"?